### DISCUSSION TITLE
ESP32 machine.WDT

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#10836] [ESP32] [answered]
What is the maximum watchdog period?  I saw some comments about fixing it in the ESP-IDF to run for more than an hour.  I am up to about 80 minutes and was wondering when I was going to hit the limit.

I assume that the WDT uses the RTC timer when in deepsleep().

**Comment by bulletmark** [comment, marked as answer]
The maximum is infinite of course (don't enable the WDT). I am certainly not a MP expert but given the timeout arg is in millisecs then I would expect the maximum time to be 2^32 / 1000 / 60 / 60  / 24 = 49 days which is obviously kind of pointless however.

**Comment by davefes** [comment]
I vaguely recall a 32 bit unsigned int but could not find the reference.  Thank you for clarifying that.

**Comment by IhorNehrutsa** [comment]
docs/ESP32: Reinitialize watchdog timer with longer timeout. #11981

**Comment by karfas** [comment]
> I assume that the WDT uses the RTC timer when in deepsleep().

There is no watchdog active during deepsleep(). I think we already discussed this in #10498.

**Reply by davefes** [reply]
Thank you.
